### ISSUE TITLE
[java] NullAssignment - FP with ternay and null as constructor argument

### ISSUE THREAD

**Issue by Vampire** [pmd/pmd#1669]
**Affects PMD Version:** *6.11.0*
**Rule:** [NullAssignment](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#nullassignment)
**Description:**
Should below code really trigger the `NullAssignment` rule?
Of course I could write `"null"` instead of `null` but why should this be worse than wasting a string constant pool place?
**Code Sample demonstrating the issue:**

```java
throw new IllegalArgumentException(String.format(
        "Unsupported client with class '%s' given",
        (discordClient == null) ? null : discordClient.getClass()));
```

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
Sounds like a FP… the rule has some checks around ternaries, but I'm unsure at first glance what scenarios they are addressing…

I guess some further research is in place.

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
